network.go

```diff
@@ -198,6 +199,11 @@ func setupRoutes(netHandle *netlink.Handle, routes []hyper.Route) error {
 			}
 		}
 
+		// Only save the routes that we are actually adding. Don't add
+		// skipped routes already existing, otherwise it could cause
+		// issues when trying to remove them.
+		finalRouteList = append(finalRouteList, route)
```

Comment:
dumb questions, you are returning `finalRouteList` only in the case  of a successful setup. But later if `removeRoutes` is called to remove (clean up) the added interfaces. It will try to remove all of them and not only the ones that were added.

So questions:
1. It is need to return the `finalRouteList` even on fail ?
2. In case being need to return finalRouteList or just to have according to the comment (`Only save the routes that we are actually adding`), can you move `finalRouteList = append(finalRouteList, route)` after call `netHandle.RouteReplace`

Reply:
Good points, I am gonna make some changes.

Reply:
looks good, thanks !